packages/flutter/lib/src/widgets/animated_list.dart

```diff
@@ -285,6 +285,36 @@ class AnimatedList extends StatefulWidget {
        assert(initialItemCount != null && initialItemCount >= 0),
        super(key: key);
 
+  /// Creates a scrolling container with separators that animates items and
+  /// separators when they are inserted or removed.
+  AnimatedList.separated({
+    Key? key,
+    required AnimatedListItemBuilder itemBuilder,
+    required AnimatedListItemBuilder separatorBuilder,
+    this.initialItemCount = 0,
+    this.scrollDirection = Axis.vertical,
+    this.reverse = false,
+    this.controller,
+    this.primary,
+    this.physics,
+    this.shrinkWrap = false,
+    this.padding,
+    this.clipBehavior = Clip.hardEdge,
+  })  : assert(itemBuilder != null),
+        assert(separatorBuilder != null),
+        assert(initialItemCount != null && initialItemCount >= 0),
+        itemBuilder =
+        ((BuildContext context, int index, Animation<double> animation) =>
+            Flex(
+              direction: scrollDirection,
+              children: <Widget>[
+                itemBuilder(context, index, animation),
+                separatorBuilder(context, index, animation),
```

Comment:
This is also different from the `ListView.separated` constructor in that the itemCount will now include the separators and the items instead of just the items. This might be confusing for people familiar with that API.

Reply:
okay, I'll try to fix this asap.